sdk/servicebus/Azure.Messaging.ServiceBus/src/Amqp/AmqpSender.cs

```diff
@@ -210,12 +207,13 @@ internal class AmqpSender : TransportSender
             ServiceBusMessageBatch messageBatch,
             CancellationToken cancellationToken)
         {
-            AmqpMessage messageFactory() => AmqpMessageConverter.BatchSBMessagesAsAmqpMessage(messageBatch.AsEnumerable<ServiceBusMessage>());
-            await _retryPolicy.RunOperation(async (timeout) =>
-                await SendBatchInternalAsync(
-                    messageFactory,
+            await _retryPolicy.RunOperation(async (sender, msgs, timeout, token) =>
+                await sender.SendBatchInternalAsync(
+                    msgs,
                     timeout,
-                    cancellationToken).ConfigureAwait(false),
+                    token).ConfigureAwait(false),
+            this,
+            messageBatch.AsEnumerable<ServiceBusMessage>(),
```

Comment:
I was fairly certain this is an OK change since the batch has a list internally. I couldn't see a reason why it would be required to cast to IEnumerable over and over again during retries. I preserved though the conversion to the AmqpMessage which seems to be the crucial part